geomstats/geometry/manifold.py

```diff
@@ -264,3 +274,161 @@ def projection(self, point):
             return gs.copy(point)
 
         raise NotImplementedError("`projection` is not implemented yet")
+
+
+class _QuotientStructureRegistry:
+    """Registry for quotient structures."""
+
+    STRUCTURES = {}
+
+    @classmethod
+    def _as_key(self, Obj):
+        """Transform an instance of a class into a key type.
```

Comment:
What is a "key type"? Could you give an example or explanation in the docstring?
For future contributors

Reply:
Not a good name. It refers to the key that is used in the `STRUCTURES` `dict`. I'll update the docstrings.